core/src/Kokkos_NumericTraits.hpp

```diff
@@ -306,6 +306,30 @@ struct reduction_identity<signed char> {
   }
 };
 
+template <>
+struct reduction_identity<bool> {
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool sum() {
+    return static_cast<bool>(false);
+  }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool prod() {
+    return static_cast<bool>(true);
+  }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool max() { return false; }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool min() { return true; }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool bor() {
+    return static_cast<bool>(false);
```

Comment:
we probably don't need the cast here :-)

Reply:
Good point.  2e4d16a drops the cast.